File: DSA/DSA06010 CÁC CHỮ SỐ XUẤT HIỆN TRONG DÃY.cpp
#include <bits/stdc++.h>
using namespace std;

void submit()
{
    int n;
    cin >> n;
    string s;
    cin.ignore();
    getline(cin, s);

    vector<bool> check(11, false);
    for (auto &&i : s)
    {
        if (isdigit(i)) check[i-'0'] = true;
    }
    
    for (int i = 0; i < 11; i++)
    {
        if (check[i])
        {
            cout << i << " ";
        }
    }
    


    cout << endl;
}


int main()
{
    ios::sync_with_stdio(false);                                    // Speed up input/output
    cin.tie(nullptr);
    cout.tie(nullptr);
    
    int so_bo_test;
    cin >> so_bo_test;
    cin.ignore();
    
    while (so_bo_test--)
    {
        submit();
    }

    return 0;
}

File: DSA_python/DSA06010 CÁC CHỮ SỐ XUẤT HIỆN TRONG DÃY.py
def submit():
    n = int(input())
    s = str(input())
    
    check = [False]*11
    for i in s:
        if i.isdigit() == True:
            check[ord(i)-ord('0')] = True
    
    for i in range(11):
        if check[i] == True:
            print(i, end=" ")
    print()



test = int(input())
for i in range(test):
    submit()
